bundle/v2/client.go

```diff
@@ -0,0 +1,581 @@
+		return nil, err
+	}
+
+	c, err := clientcache.New(conf.CacheDir, conf.TempDir)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := baseClient.StdHTTPClient() // Bidi streams don't work with retryable HTTP client.
+	return &Client{
+		Client:    baseClient,
+		rpcClient: bundlev2connect.NewCerbosBundleServiceClient(httpClient, baseClient.APIEndpoint, options...),
+		cache:     c,
+	}, nil
+}
+
+func (c *Client) BootstrapBundle(ctx context.Context, bundleLabel string) (string, error) {
+	log := c.Logger.WithValues("bundle", bundleLabel)
+	log.V(1).Info("Getting bootstrap configuration")
+
+	bootstrapConfName := hex.EncodeToString(
+		credentials.Hash(
+			c.Credentials.BootstrapKey,
+			[]byte(bundleLabel),
+		),
+	)
+	bootstrapURL, err := url.JoinPath(c.BootstrapEndpoint, bootstrapV2PathPrefix, c.Credentials.ClientID, bootstrapConfName)
+	if err != nil {
+		return "", fmt.Errorf("failed to construct bootstrap URL: %w", err)
+	}
+
+	bootstrapConf, err := c.downloadBootstrapConf(ctx, bootstrapURL)
+	if err != nil {
+		log.Error(err, "Failed to download bootstrap configuration")
+		return "", err
+	}
+
+	if meta := bootstrapConf.Meta; meta != nil {
+		log.Info("Bootstrap configuration downloaded", "created_at", meta.CreatedAt.AsTime(), "commit_hash", meta.CommitHash)
+	}
+
+	base.LogResponsePayload(log, bootstrapConf)
+	return c.getBundleFile(logr.NewContext(ctx, log), bootstrapConf.BundleInfo)
+}
+
+func (c *Client) downloadBootstrapConf(ctx context.Context, url string) (*bootstrapv2.PDPConfig, error) {
+	log := logr.FromContextOrDiscard(ctx).WithValues("url", url)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		log.V(1).Error(err, "Failed to construct download request")
+		return nil, fmt.Errorf("failed to construct download request: %w", err)
+	}
+
+	log.V(1).Info("Sending download request")
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		log.V(1).Error(err, "Failed to send download request")
+		return nil, fmt.Errorf("failed to send download request: %w", err)
+	}
+
+	log.V(1).Info(fmt.Sprintf("Download request status: %s", resp.Status))
+	defer func() {
+		if resp.Body != nil {
+			_, _ = io.Copy(io.Discard, resp.Body)
+			_ = resp.Body.Close()
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusNotFound {
+			log.V(1).Info("Bootstrap bundle not found")
+			return nil, bundle.ErrBootstrapBundleNotFound
+		}
+
+		log.V(1).Info("Failed to download bootstrap bundle")
+		return nil, bundle.ErrDownloadFailed
+	}
+
+	conf, err := c.Credentials.DecryptV2(io.LimitReader(resp.Body, bundle.MaxBootstrapSize))
+	if err != nil {
+		log.V(1).Error(err, "Failed to decrypt bootstrap bundle")
+		return nil, fmt.Errorf("failed to decrypt bootstrap bundle: %w", err)
+	}
+
+	return c.parseBootstrapConf(conf)
+}
+
+func (c *Client) parseBootstrapConf(conf []byte) (*bootstrapv2.PDPConfig, error) {
+	trimmed := bytes.TrimLeftFunc(conf, unicode.IsSpace)
```

Comment:
Why do we need to trim or check for JSON? We know that the contents of the bootstrap bundle info will be a `VTMarshal`ed protobuf message, don't we?